cmd/oras/pull.go

```diff
@@ -125,9 +121,7 @@ func runPull(opts pullOptions) error {
 			return err
 		}
 	}
-	if targetPlatform != nil {
-		copyOptions.WithTargetPlatform(targetPlatform)
-	}
+	copyOptions.WithTargetPlatform(opts.OCIPlatform)
```

Comment:
nit: Although `WithTargetPlatform` supports `nil` input, it is better to have an `if` statement just for readability.

Reply:
Should we also enforce the nil check for fetch option and resolve option?